server/web/routes/proposals.js

```diff
@@ -322,6 +323,51 @@ async function getObjectFromS3(fileName) {
   });  
 }
 
+async function findAllProposalsInSameGroup(proposal, includeSelf = false) {
+  const history = [];
+  let original = null;
+  if (proposal.hasChild || proposal.parentId) {
+    // find all revised proposals
+    const revisedProposal = await Proposal.find({
+      groupId: proposal.groupId ? proposal.groupId : proposal._id.toString(),
+    });
+
+    if (proposal.groupId) {
+      // if current proposal is a revised proposal, add the original proposal to the list
+      original = await Proposal.findById(proposal.groupId);
+    } else {
+      // if current proposal is the original proposal, add itself to the list
+      original = proposal;
+    }
+
+    for (const prop of revisedProposal) {
+      if (!includeSelf && prop._id.toString() === proposal._id.toString()) continue;
+
+      history.push({
+        id: prop._id.toString(),
+        fileName: prop.fileName,
+        date: prop.createdAt,
+        dateString: prop.createdAt.toDateString(),
+      });
+    }
+
+    history.sort((a, b) => {
+      return b.date - a.date;
+    });
+  } else original = proposal;
```

Comment:
We don't need this since we removed the outer if condition and in line 337 and 340 you have already found the original proposal.So remove the else statement as well.